lang/luasocket/Makefile

```diff
@@ -8,33 +8,59 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=luasocket
-PKG_SOURCE_VERSION:=6d5e40c324c84d9c1453ae88e0ad5bdd0a631448
-PKG_VERSION:=3.0-rc1-20130909
-PKG_RELEASE:=5
+PKG_SOURCE_DATE:=2019-04-21
+PKG_SOURCE_VERSION:=733af884f1aa18ff469bf3c4d18810e815853211
+PKG_RELEASE:=1
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
-PKG_MIRROR_HASH:=d2fa075d8bd026c41e0eb1a634ac2ad8115dee8abb070720e8e91fab51f86ee4
-PKG_SOURCE_URL:=https://github.com/diegonehab/luasocket.git
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE_URL:=https://github.com/diegonehab/luasocket
+PKG_MIRROR_HASH:=60aef7544426cae3e6c7560a6e4ad556a04b879ca0ad0311645b2c513c872128
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(BUILD_VARIANT)/$(PKG_NAME)-$(PKG_SOURCE_DATE)-$(call version_abbrev,$(PKG_SOURCE_VERSION))
+
+PKG_MAINTAINER:=W. Michael Petullo <mike@flyn.org>
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
 
 include $(INCLUDE_DIR)/package.mk
 
-define Package/luasocket
+define Package/luasocket/default
   SUBMENU:=Lua
   SECTION:=lang
   CATEGORY:=Languages
+  URL:=http://w3.impa.br/~diego/software/luasocket
+endef
+
+define Package/luasocket
+  $(Package/luasocket/default)
   TITLE:=LuaSocket
-  URL:=http://luasocket.luaforge.net/
-  MAINTAINER:=W. Michael Petullo <mike@flyn.org>
   DEPENDS:=+lua
+  VARIANT:=five-one
+  DEFAULT_VARIANT:=1
+endef
+
+define Package/luasocket-5.3
+  $(Package/luasocket/default)
+  TITLE:=LuaSocket 5.3
+  DEPENDS:=+liblua5.3
+  VARIANT:=five-three
 endef
 
-define Package/luasocket/description
+ifeq ($(BUILD_VARIANT),five-one)
+  LUA_VERSION=5.1
+endif
+
+ifeq ($(BUILD_VARIANT),five-three)
+  LUA_VERSION=5.3
+endif
+
+
+define Package/luasocket/default/description
   LuaSocket is the most comprehensive networking support
   library for the Lua language. It provides easy access to
   TCP, UDP, DNS, SMTP, FTP, HTTP, MIME and much more.
 endef
+Package/luasocket/description     = $(Package/luasocket/default/description)
```

Comment:
iirc, you can't actually have a variant match the base name. It appears to work in local testing, but it all falls apart on buildbots.  this is why we often have -nossl and -ssl rather than "" (no suffix) and "-ssl" for the ssl variants. Unless you've confirmed with jow on irc that variants matching the base name are ok, please fix this.  Note how this plays in the "depends" clauses, you need to make sure that you are _not_ providing things tha tmatch depends, becuase these are _not_compatible.

Reply:
@karlp - Again, I'm very sorry, I don't understand what you're asking for.  To get the description stuff I copied how another makefile was doing it.  Are you suggesting I just change all the luasocket package strings to luasocket-5.1?

Reply:
I discussed this with @karlp on IRC and now that we have `VARIANT:=lua-51` and `VARIANT:=lua-53` there should no conflict with base package name anymore.